pkg/internal/resources/api-server/apiserver.go

```diff
@@ -32,8 +35,11 @@ import (
 type Config struct {
 	// Namespace is the KubeCarrier master controller manager should be deployed into.
 	Namespace string
-	// Name of this Kubecarrier API object.
+	// Name of this KubeCarrier API object.
 	Name string
+
+	// Spec of the APIServer
+	Spec operatorv1alpha1.APIServerSpec
```

Comment:
Why don't you just pass the secret name here?

Reply:
Because many other stuff shall be needed in the future and I'll rather pass the whole spec instead of writing glue code which translates stuff around

Reply:
Please add the glue code here.

I would like to have the code in here just executing the template, while the controller needs to prepare/default fields as needed. Coupling this directly to the API structs is just lazy.